src/middlewares.rs

```diff
@@ -18,30 +21,72 @@ pub struct UserState {
 
 pub type CookieUserState = Extension<UserState>;
 
+impl<'a> TryInto<Cookie<'a>> for UserState {
+    type Error = AppError;
+
+    fn try_into(self) -> Result<Cookie<'a>, Self::Error> {
+        let new_state_json = serde_json::to_string(&self).map_err(|err| {
```

Comment:
Just curious, when can this come up? I'm not sure that it should be possible to create a `UserState` that can't be serialized, as that would constitute a bug in our program, so I don't think returning a 400 status code is appropriate here.

The two options I'd recommend are either set `type Error = anyhow::Error` and consider this a fatal unhandleable error, or just use `.expect()` to unwrap the error and panic otherwise.

It seems like this would essentially be a bug in our program if we can't serialize the user state, so maybe we should use `.expect()` here and panic instead.

The second options is personally appealing because it would be incredibly convenient to use with CookieJar, since [`CookieJar::add`](https://docs.rs/axum-extra/latest/axum_extra/extract/cookie/struct.CookieJar.html#method.add) takes an `Into<Cookie>`, but idk, it might be good to catch the error instead, your call. I'm still personally trying to decide when it's appropriate to panic or not.